### PR TITLE
Fix buffering issues with xcpretty.

### PR DESCRIPTION
**Pull request by mtauraso**
This makes the output update more smoothly :)

### CODE REVIEW

File: lib/xcpretty/printer.rb [hunk 1]
```diff
@@ -13,7 +13,10 @@ def initialize(params)
 
     def pretty_print(text)
       formatted_text = formatter.pretty_format(text)
-      STDOUT.print(formatted_text + formatter.optional_newline) unless formatted_text.empty?
+      unless formatted_text.empty?
+        STDOUT.print(formatted_text + formatter.optional_newline)
+        STDOUT.flush()
```
Comment (mtauraso) [inline lib/xcpretty/printer.rb]: gotcha.

**Comment by supermarin** [comment]
@mtauraso thanks for the contribution!

Do you mind posting a screenshot before/after, just documentation-wise?

**Comment by supermarin** [comment]
@mtauraso /ping

I've tried to run with or without .flush and can't see any difference.
Could you please post a screenshot?

**Comment by xfreebird** [comment]
We always have this buffering issues when run from Jenkins with tee. It would take about 30 seconds to get some output.

You can test that by running a job from Jenkins connected through ssh to a remote Slave by running the command below: 
xcodebuild -workspace 'ClientApp.xcworkspace' -scheme 'ClientApp' -configuration 'Release' -sdk 'iphoneos' build   | tee 'artifacts/xcodebuild.log' | xcpretty 

I tried this patch, and it helps. Now all looks like in real time.
Thanks @mtauraso !

@supermarin please merge this as it would fix this buffering issues.

**Comment by supermarin** [comment]
@xfreebird oh i see, it totally makes sense. thanks!

**Pull request merged**